pkg/util/sys/device.go

```diff
@@ -399,17 +409,83 @@ func parseUdevInfo(output string) map[string]string {
 	return result
 }
 
-func isDeviceAvailable(executor exec.Executor, devicePath string) (bool, string, error) {
-	CVInventory, err := inventoryDevice(executor, devicePath)
+func isDeviceAvailable(executor exec.Executor, devicePath string) (available bool, rejectedReasons string, outErr error) {
+	output, err := executor.ExecuteCommandWithOutput("lsblk", devicePath,
+		"--bytes", "--noheadings", "--output", "RM,RO,SIZE,TYPE")
 	if err != nil {
-		return false, "", fmt.Errorf("failed to determine if the device %q is available. %v", devicePath, err)
+		return false, "", errors.Wrapf(err, "failed to determine if device %q is available", devicePath)
+	}
+	items := strings.Fields(output)
+	if len(items) != 4 {
+		return false, "", errors.Errorf(
+			"failed to determine if device %q is available. lsblk output should have 4 fields but has %d: %s", devicePath, len(items), items)
 	}
+	rm := items[0]
+	ro := items[1]
+	sizeStr := items[2]
+	devType := items[3]
 
-	if CVInventory.Available {
-		return true, "", nil
+	rejectedList := []string{}
+
+	// must not be removable
+	if rm != "0" { // 0 means not removable, 1 means removable
```

Comment:
I had doubts about this one but after looking at some kernel code from https://elixir.bootlin.com/linux/latest/source/include/linux/genhd.h#L96:

 ```
 * ``GENHD_FL_REMOVABLE`` (0x0001): indicates that the block device
 * gives access to removable media.
 * When set, the device remains present even when media is not
 * inserted.
 * Must not be set for devices which are removed entirely when the
 * media is removed.
 ```

Reply:
Ceph-volume gets this param from `/sys/block/sdX` or from `/sys/block/sdX/sdXP` if it's a partition. Are you thinking that lsblk might report this differently from `/sys/block`?